src/mame/barcrest/mpu1.cpp

```diff
@@ -0,0 +1,518 @@
+	PORT_DIPNAME( 0x01, 0x01, "Gamble feature limit" )
+	PORT_DIPSETTING(    0x00, "£1" )
+	PORT_DIPSETTING(    0x01, "50p" )
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_POKER_HOLD1 ) PORT_NAME("Match 1/Gamble")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_POKER_HOLD2 ) PORT_NAME("Match 2")
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_POKER_HOLD3 ) PORT_NAME("Match 3")
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_GAMBLE_PAYOUT ) PORT_NAME("Collect")
+INPUT_PORTS_END
+
+static INPUT_PORTS_START( m_mtchup )
+	PORT_INCLUDE( mpu1_inputs )
+
+	PORT_MODIFY("IN")
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_POKER_HOLD1 ) PORT_NAME("Match 1/Gamble")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_POKER_HOLD2 ) PORT_NAME("Match 2")
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_POKER_HOLD3 ) PORT_NAME("Match 3")
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_GAMBLE_PAYOUT ) PORT_NAME("Collect")
+
+	PORT_MODIFY("COIN")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_UNUSED )
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_COIN1 ) PORT_NAME("10p") PORT_CHANGED_MEMBER(DEVICE_SELF, mpu1_state, coin_input, 1) PORT_IMPULSE(2)
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_COIN2 ) PORT_NAME("10p Token") PORT_CHANGED_MEMBER(DEVICE_SELF, mpu1_state, coin_input, 2) PORT_IMPULSE(2)
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_COIN3 ) PORT_NAME("50p") PORT_CHANGED_MEMBER(DEVICE_SELF, mpu1_state, coin_input, 3) PORT_IMPULSE(2)
+INPUT_PORTS_END
+
+static INPUT_PORTS_START( m_lndg )
+	PORT_INCLUDE( mpu1_inputs )
+
+	PORT_MODIFY("IN")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_NAME("Respin")
+	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_BUTTON2 ) PORT_NAME("Gamble")
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_GAMBLE_PAYOUT ) PORT_NAME("Collect")
+INPUT_PORTS_END
+
+void mpu1_state::machine_start()
+{
+	m_lamps.resolve();
+	m_reel_out.resolve();
+	m_sreel_out.resolve();
+
+	save_item(NAME(m_reel_state));
+	save_item(NAME(m_reel_pos));
+	save_item(NAME(m_reel_speed));
+
+	m_reel_timer[0] = timer_alloc(FUNC(mpu1_state::reel_move<0>), this);
+	m_reel_timer[1] = timer_alloc(FUNC(mpu1_state::reel_move<1>), this);
+	m_reel_timer[2] = timer_alloc(FUNC(mpu1_state::reel_move<2>), this);
+	m_reel_timer[3] = timer_alloc(FUNC(mpu1_state::reel_move<3>), this);
+	m_change_pia2a_bit7_timer = timer_alloc(FUNC(mpu1_state::change_pia2a_bit7), this);
+
+	for(int i = 0; i < 4; i++) 
+	{
+		m_reel_state[i] = REEL_STOPPED;
+		m_reel_pos[i] = 0;
+	}
+}
+
+void mpu1_state::machine_reset()
+{
+	// Return PIA2 port A bit 7 as set for 0.1s from boot, then clear it
+	m_pia2a_bit7_value = 0x80;
+	m_change_pia2a_bit7_timer->adjust(attotime::from_msec(100));
+}
+
+void mpu1_state::mpu1(machine_config &config)
+{
+	M6800(config, m_maincpu, 1_MHz_XTAL); /* On MPU1, the clock comes from a multivibrator circuit varying somewhere around 1 MHz from
+	                                         board to board. This results in for example slightly different sound pitch across machines.
+	                                         MPU2 replaced this with a crystal for a stable clock, which I'm using here. */
```

Comment:
You shouldn’t use the `_XTAL` thing for things that are not crystals/resonators.  Just specify a frequency, like `1'000'000`.

Reply:
Fixed.